### PR TITLE
chore(DTFS2-6999): replace moment in tfm-ui

### PR DESCRIPTION
**Pull request by MarRobSoftwire**
## Introduction :pencil2:
Moment js is deprecated, it should be replaced. This PR will not cover removing moment from tests - see #2761 .

## Resolution :heavy_check_mark:
* Replace moment() with vanilla Dates where possible and date-fns where necessary in trade-finance-manager-ui

## Misc
- add `dev` test scripts to all packages

### CODE REVIEW

File: trade-finance-manager-ui/component-tests/case/activity/_macros/deal-submitted.component-test.js [hunk 1]
```diff
@@ -34,8 +34,8 @@ describe(component, () => {
   });
 
   it('should render submissionDate', () => {
-    const expectedDate = localiseTimestamp(params.submissionDate, 'D MMMM YYYY', params.userTimezone);
-    const expectedTime = localiseTimestamp(params.submissionDate, 'HH:mma', params.userTimezone);
+    const expectedDate = filterLocaliseTimestamp(params.submissionDate, 'd MMMM yyyy', params.userTimezone);
+    const expectedTime = filterLocaliseTimestamp(params.submissionDate, 'HH:mmaaa', params.userTimezone);
```
Comment (MarRobSoftwire) [inline trade-finance-manager-ui/component-tests/case/activity/_macros/deal-submitted.component-test.js:38]: Sanity check please:
|[moment](https://momentjscom.readthedocs.io/en/latest/moment/04-displaying/01-format/) |  [date-fns](https://date-fns.org/v3.3.1/docs/format)| example|
|---------|--------|---|
|`h` | `h`| 1, 2, 3, ... 11, 12 |
|`mm`| `mm`| 00, 01, ..., 58, 59 |
|`a`|`aaa`| am, pm |

File: trade-finance-manager-ui/package.json [hunk 2]
```diff
@@ -58,8 +60,7 @@
     "joi": "^17.12.1",
     "jquery": "3.6.4",
     "lodash.isequal": "^4.5.0",
-    "moment": "2.30.1",
-    "moment-timezone": "^0.5.44",
+    "moment": "^2.30.1",
```
Comment (MarRobSoftwire) [inline trade-finance-manager-ui/package.json:63]: Note to uninstall

File: trade-finance-manager-ui/component-tests/case/activity/_macros/deal-submitted.component-test.js (hunk 1 above)
Comment (AlexBramhill) [reply trade-finance-manager-ui/component-tests/case/activity/_macros/deal-submitted.component-test.js:38]: lgtm

File: trade-finance-manager-ui/server/nunjucks-configuration/filter-formatDateString.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { format, isValid, parse } from "date-fns";
+
+/**
+ * @param dateStr date string formatted as fromFormat
+ * @param fromFormat date format using {@link https://date-fns.org/v3.3.1/docs/format | unicode tokens}
+ * @param toFormat date format using {@link https://date-fns.org/v3.3.1/docs/format | unicode tokens}
+ * @returns date formatted as toFormat or 'Invalid date' if can't parse dateStr
+ */
+export const filterFormatDateString = (dateStr: string, fromFormat: string, toFormat: string = 'd MMM yyyy') => {
```
Comment (AlexBramhill) [inline trade-finance-manager-ui/server/nunjucks-configuration/filter-formatDateString.ts:9]: I really hate this name, can we change it back please

**Review by AlexBramhill — COMMENTED** [review]
lgtm with one comment

**Review by AlexBramhill — APPROVED** [review]

**Pull request merged**